test/Confluent.Kafka.IntegrationTests/Tests/AssignPastEnd.cs

```diff
@@ -23,12 +23,12 @@
 
 namespace Confluent.Kafka.IntegrationTests
 {
-    /// <summary>
-    ///     Test functionality of Consumer.Consume when assigned to offest
-    ///     higher than the offset of the last message on a partition.
-    /// </summary>
     public static partial class Tests
     {
+        /// <summary>
+        ///     Test functionality of Consumer.Consume when assigned to offest
+        ///     higher than the offset of the last message on a partition.
+        /// </summary>
```

Comment:
ahh ok, i see that error is an option for auto.offset.reset now and also see why this is an intuitive way for this to work.

note that lastMsg+2 is the first offset to be considered out of range (makes sense), and given your other comments in this PR about what offset is, of course lastMsg+1 will be in range (I didn't properly get that before).